Whitelisting scala.sys.SystemProperties crashes the compiler.

Whitelisting `scala.sys.SystemProperties` will crash the compiler when running `sbt test` and the  
errors are:
```scala
[error] Test dotty.tools.dottydoc.TestWhitelistedCollections.arrayHasDocumentation failed: java.lang.AssertionError: assertion failed, took 8.677 sec
[error]     at scala.Predef$.assert(Predef.scala:151)
[error]     at dotty.tools.dotc.core.Types$TypeBounds.<init>(Types.scala:3091)
[error]     at dotty.tools.dotc.core.Types$TypeAlias.<init>(Types.scala:3162)
[error]     at dotty.tools.dotc.core.Types$CachedTypeAlias.<init>(Types.scala:3188)
[error]     at dotty.tools.dotc.core.Uniques$TypeAliasUniques.newAlias$1(Uniques.scala:84)
[error]     at dotty.tools.dotc.core.Uniques$TypeAliasUniques.enterIfNew(Uniques.scala:89)
[error]     at dotty.tools.dotc.core.Types$TypeAlias$.apply(Types.scala:3202)
[error]     at dotty.tools.dotc.core.Types$Type.bounds(Types.scala:1128)
[error]     at dotty.tools.dottydoc.core.DocASTPhase.dotty$tools$dottydoc$core$DocASTPhase$$membersFromSymbol$1(DocASTPhase.scala:47)
```

```scala
[error] Test dotty.tools.dottydoc.TestWhitelistedCollections.traitImmutableHasDocumentation failed: java.lang.AssertionError: assertion failed, took 6.213 sec
[error]     at scala.Predef$.assert(Predef.scala:151)
[error]     at dotty.tools.dotc.core.Types$TypeBounds.<init>(Types.scala:3091)
[error]     at dotty.tools.dotc.core.Types$TypeAlias.<init>(Types.scala:3162)
[error]     at dotty.tools.dotc.core.Types$CachedTypeAlias.<init>(Types.scala:3188)
[error]     at dotty.tools.dotc.core.Uniques$TypeAliasUniques.newAlias$1(Uniques.scala:84)
[error]     at dotty.tools.dotc.core.Uniques$TypeAliasUniques.enterIfNew(Uniques.scala:89)
[error]     at dotty.tools.dotc.core.Types$TypeAlias$.apply(Types.scala:3202)
[error]     at dotty.tools.dotc.core.Types$Type.bounds(Types.scala:1128)
[error]     at dotty.tools.dottydoc.core.DocASTPhase.dotty$tools$dottydoc$core$DocASTPhase$$membersFromSymbol$1(DocASTPhase.scala:47)
```

--- a/compiler/test/dotc/scala-collections.blacklist
+++ b/compiler/test/dotc/scala-collections.blacklist
@@ -222,8 +222,5 @@
 #    |  result type of implicit definition needs to be given explicitly
 
 
-../scala-scala/src/library/scala/sys/SystemProperties.scala
-# crashes on dotty.tools.dottydoc.TestWhitelistedCollections.arrayHasDocumentation
-
 ../scala-scala/src/library/scala/util/Sorting.scala
 # assertion failed: invalid prefix ImplicitMethodType(List(ord), List(RefinedType(TypeRef(ThisType(TypeRef(NoPrefix,math)),Ordering), scala$math$Ordering$$T, TypeAlias(TypeRef(NoPrefix,K), 0))), RefinedType(TypeRef(ThisType(TypeRef(NoPrefix,math)),Ordering), scala$math$Ordering$$T, TypeAlias(TypeRef(NoPrefix,K), 0)))

--- a/compiler/test/dotc/scala-collections.whitelist
+++ b/compiler/test/dotc/scala-collections.whitelist
@@ -386,6 +386,7 @@
 
 ../scala-scala/src/library/scala/compat/Platform.scala
 
+../scala-scala/src/library/scala/sys/SystemProperties.scala
 ../scala-scala/src/library/scala/sys/package.scala
 ../scala-scala/src/library/scala/sys/Prop.scala
 ../scala-scala/src/library/scala/sys/PropImpl.scala

--- a/doc-tool/src/dotty/tools/dottydoc/core/DocASTPhase.scala
+++ b/doc-tool/src/dotty/tools/dottydoc/core/DocASTPhase.scala
@@ -8,6 +8,7 @@ import dotc.CompilationUnit
 import dotc.config.Printers.dottydoc
 import dotc.core.Contexts.Context
 import dotc.core.Comments.ContextDocstrings
+import dotc.core.Types.NoType
 import dotc.core.Phases.Phase
 import dotc.core.Symbols.{ Symbol, NoSymbol }
 
@@ -44,32 +45,35 @@ class DocASTPhase extends Phase {
     }
 
     def membersFromSymbol(sym: Symbol): List[Entity] = {
-      val defs = sym.info.bounds.hi.membersBasedOnFlags(Flags.Method, Flags.Synthetic | Flags.Private)
-        .filterNot(_.symbol.owner.name.show == "Any")
-        .map { meth =>
-          DefImpl(
-            meth.symbol,
-            meth.symbol.name.show,
-            Nil,
-            path(meth.symbol),
-            returnType(meth.info),
-            typeParams(meth.symbol),
-            paramLists(meth.info),
-            implicitlyAddedFrom = Some(returnType(meth.symbol.owner.info))
+      if (sym.info ne NoType) {
+        val defs = sym.info.bounds.hi.membersBasedOnFlags(Flags.Method, Flags.Synthetic | Flags.Private)
+          .filterNot(_.symbol.owner.name.show == "Any")
+          .map { meth =>
+            DefImpl(
+              meth.symbol,
+              meth.symbol.name.show,
+              Nil,
+              path(meth.symbol),
+              returnType(meth.info),
+              typeParams(meth.symbol),
+              paramLists(meth.info),
+              implicitlyAddedFrom = Some(returnType(meth.symbol.owner.info))
+            )
+          }.toList
+
+        val vals = sym.info.fields.filterNot(_.symbol.is(Flags.Private | Flags.Synthetic)).map { value =>
+          ValImpl(
+            value.symbol,
+            value.symbol.name.show,
+            Nil, path(value.symbol),
+            returnType(value.info),
+            implicitlyAddedFrom = Some(returnType(value.symbol.owner.info))
           )
-        }.toList
+        }
 
-      val vals = sym.info.fields.filterNot(_.symbol.is(Flags.Private | Flags.Synthetic)).map { value =>
-        ValImpl(
-          value.symbol,
-          value.symbol.name.show,
-          Nil, path(value.symbol),
-          returnType(value.info),
-          implicitlyAddedFrom = Some(returnType(value.symbol.owner.info))
-        )
+        defs ++ vals
       }
-
-      defs ++ vals
+      else Nil
     }
 
 

--- a/doc-tool/src/dotty/tools/dottydoc/model/factories.scala
+++ b/doc-tool/src/dotty/tools/dottydoc/model/factories.scala
@@ -9,6 +9,7 @@ import dotc.core.TypeApplications._
 import dotc.core.Contexts.Context
 import dotc.core.Symbols.{ Symbol, ClassSymbol }
 import dotty.tools.dotc.core.SymDenotations._
+import dotty.tools.dotc.config.Printers.dottydoc
 import dotty.tools.dotc.core.Names.TypeName
 import dotc.ast.Trees._
 
@@ -105,6 +106,17 @@ object factories {
           else paramName
 
         typeRef(name)
+      case tr: TermRef =>
+        /** A `TermRef` appears in the return type in e.g:
+          * ```
+          * def id[T](t: T): t.type = t
+          * ```
+          */
+        val name = tr.show
+        if (!name.endsWith(".type"))
+          ctx.warning(s"unhandled return type found: $tr")
+
+        typeRef(name, params = params)
     }
 
     expandTpe(t)
@@ -154,7 +166,8 @@ object factories {
 
     case annot: AnnotatedType => paramLists(annot.tpe)
     case (_: PolyParam | _: RefinedType | _: TypeRef | _: ThisType |
-          _: ExprType  | _: OrType      | _: AndType | _: HKApply) => Nil // return types should not be in the paramlist
+          _: ExprType  | _: OrType      | _: AndType | _: HKApply  | _: TermRef) =>
+      Nil // return types should not be in the paramlist
   }
 
   def superTypes(t: Tree)(implicit ctx: Context): List[MaterializableLink] = t.symbol.denot match {

